### ISSUE TITLE
Dialog A11y broken when title id is passed in

### ISSUE THREAD

**Issue by cabillin** [microsoft/fluentui#18000]
### Environment Information

- **Package version(s)**: repros in @fluentui/react@8, but we are using ^7.137.2
- **Browser and OS versions**: edge dev (but browser agnostic)

### Describe the issue:

div with role="alertdialog" does not receive the correct id when title id is passed into titleProps

![image](https://user-images.githubusercontent.com/71525014/116635916-dd2bb000-a914-11eb-9e95-5f8233de6045.png)

Also, flagged by axe
Expected the HTML found at $('div[role="alertdialog"]') to have no violations:

    <div role="alertdialog" aria-modal="true" aria-labelledby="Dialog642-title" style="outline: none;">

    Received:

    "ARIA attributes must conform to valid values (aria-valid-attr-value)"

    Fix all of the following:
      Invalid ARIA attribute value: aria-labelledby="Dialog642-title"

    You can find more information on this issue here: 
    https://dequeuniversity.com/rules/axe/4.1/aria-valid-attr-value?application=axeAPI

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/KnetTravlr/pen/LYxovWm

#### Actual behavior:

![image](https://user-images.githubusercontent.com/71525014/116635916-dd2bb000-a914-11eb-9e95-5f8233de6045.png)

#### Expected behavior:

id passed to div with role="alertdialog" is the one passed into actual title

### Documentation describing expected behavior

common sense/lack of comment telling me I need to do something extra if I pass in my own id

**Comment by gouttierre** [comment]
@cabillin - thanks for reaching out to us and sharing the details into the repro.

@miroslavstastny - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by miroslavstastny** [comment]
This does not seem to be a regression - I can repro in 7.170.1.

There are two props which you can set:
1. `dialogContentProps.titleId` (this is v7, it has been deprecated in v8 where you should use `dialogContentProps.titleProps.id`) - this prop sets `id` attribute on the `div` inside Dialog's header.
2. `modalProps.titleAriaId` - this prop sets `aria-labelledby` attribute on the `div[role="dialog"]`.

If you want to have a custom id, you need to explicitly set both props to the same value - @microsoft/cxe-red is this expected behavior?
